src/processor/tasks.py

```diff
@@ -7,16 +7,56 @@
 
 from core import db
 from core.celery import celery as celery_app
+from core.logic import get_enum_by_value
 from core.settings import Origins, StaticProviders
 from processor.collections.reasons import doi_not_on_wikipedia_page
+from processor.exceptions import TimeoutException
 from processor.logic import check_wikipedia_event
 from processor.models import Event, RawEvent, Scrape, Uri
 
-from .utils import event_generator
 
 logger = get_task_logger(__name__)
 
 
+@celery_app.task(name='process-plugin', bind=True)
+def process_plugin(
+        self,
+        plugin_name,
+        uri_id,
+        origin_value,
+        scrape_id,
+        last_check_iso
+):
+    # Get around objects not being JSON serialisable for tasks...
+    plugin = current_app.config.get('PLUGINS').get(plugin_name)
+    uri = Uri.query.get(uri_id)
+    scrape = Scrape.query.get(scrape_id)
+    origin = get_enum_by_value(Origins, origin_value)
+    last_check = datetime.fromisoformat(last_check_iso)
+
+    try:
```

Comment:
The reason I went with a `try/except` was so we can use `TimeoutException.timeout` to specify how long to wait before retrying a task, rather than maxing out the `retry_backoff` delay; but I can change it back :+1: